frontend/catalyst/jax_primitives.py

```diff
@@ -662,6 +662,15 @@ def _grad_abstract(*args, jaxpr, fn, grad_params):
     return tuple(transformed_signature.get_results())
 
 
+def _get_call_jaxpr(jaxpr):
+    """Extracts the `call_jaxpr` from a JAXPR if it exists.""" ""
+    for eqn in jaxpr.eqns:
+        primitive = eqn.primitive
+        if primitive is func_p:
+            return eqn.params["call_jaxpr"]
+    raise AssertionError("No call_jaxpr found in the JAXPR.")
```

Comment:
Can't believe the fix was so simple 😅 

Can you confirm that the design matches what we expect? I think what would need to happen is
- the folding is applied to the quantum function generated inside the dynamic one shot function
- the fitting is applied to the entire dynamic one shot function, which includes the execution loop and post-processing

Reply:
I am not too familiar with the ZNE algorithm, but as far as I understand now, this is what happens. I can see that the extrapolation is happening at a higher level on the results of the zne_p primitive which uses the dynamic-one-shot function. I am not too sure where the folding happens but I can see that the quantum function (not the dynamic one-shot function) is used to create the lowered ZNE operation which is where I assume the folding happens. I can dig deeper more next week though :)

Reply:
Nice, just wanted to confirm. Good work :)